### PR TITLE
Fix ordering of operands in ScheduledPipelineOp::build

### PR DESCRIPTION
**Pull request by blakep-msft**
This is a fix for #5584.  `ScheduledPipelineOp::build` is changed to add operands in the same order that `PipelineOps.td` expects.

### CODE REVIEW

File: lib/Dialect/Pipeline/PipelineOps.cpp [hunk 1]
```diff
@@ -386,12 +386,14 @@ void ScheduledPipelineOp::build(OpBuilder &odsBuilder, OperationState &odsState,
                                 StringAttr name) {
   odsState.addOperands(inputs);
   odsState.addOperands(extInputs);
-  odsState.addOperands(go);
-  odsState.addOperands(clock);
-  odsState.addOperands(reset);
+
```
Comment (teqdruid) [inline lib/Dialect/Pipeline/PipelineOps.cpp:389]: nit: I would prefer no newlines here.

**Review by teqdruid — APPROVED** [review]
this looks right!

File: test/Dialect/Pipeline/Transforms/schedule-linear-pipeline.mlir [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 // CHECK-LABEL:   hw.module @pipeline(
 // CHECK-SAME:          %[[VAL_0:.*]]: i32, %[[VAL_1:.*]]: i32, %[[VAL_2:.*]]: i1, %[[VAL_3:.*]]: i1, %[[VAL_4:.*]]: i1) -> (out: i32) {
-// CHECK:           %[[VAL_5:.*]], %[[VAL_6:.*]] = pipeline.scheduled(%[[VAL_7:.*]] : i32 = %[[VAL_0]], %[[VAL_8:.*]] : i32 = %[[VAL_1]]) clock(%[[VAL_9:.*]] = %[[VAL_2]]) reset(%[[VAL_10:.*]] = %[[VAL_3]]) go(%[[VAL_11:.*]] = %[[VAL_4]]) -> (out : i32) {
+// CHECK:           %[[VAL_5:.*]], %[[VAL_6:.*]] = pipeline.scheduled(%[[VAL_7:.*]] : i32 = %[[VAL_0]], %[[VAL_8:.*]] : i32 = %[[VAL_1]]) clock(%[[VAL_9:.*]] = %[[VAL_3]]) reset(%[[VAL_10:.*]] = %[[VAL_4]]) go(%[[VAL_11:.*]] = %[[VAL_2]]) -> (out : i32) {
```
Comment (teqdruid) [inline test/Dialect/Pipeline/Transforms/schedule-linear-pipeline.mlir:5]: Why was this change necessary? This implies that this particular builder was being used by the parser.

Comment (mortbopet) [reply test/Dialect/Pipeline/Transforms/schedule-linear-pipeline.mlir:5]: The builder is used in the scheduling pass, hence when the result was being printed, operands were printed in the wrong order. If i (or the reviewers) of this had been a bit more attentive, we'd have spotted this issue there.

**Review by mortbopet — APPROVED** [review]
Awesome - thank you for spotting this and fixing it. I could say that it was something i intentionally put in here to get you to do your first PR, but then i'd be lying and covering up for my own negligence... 🙃.
